src/inputs/ObjectSearch.tsx

```diff
@@ -52,16 +101,75 @@ class ObjectSearch extends Component<IObjectSearchProps> {
       ;
 
     this.search = value;
-    this.props.onSearchChange(this.search);
+    this.isLoading.setTrue();
+    onSearchChange(this.search);
     const response = await getEndpoint(`/${endpoint}/${toKey(params)}`);
     this.options = response.results;
+    this.isLoading.setFalse();
+  }
+
+  private renderOptionAdd () {
+    const { addNewContent } = this.props
+      , className = `${SEARCH_CREATE_PREFIX_CLS}-item-add`;
+
+    return (
+      <Antd.Select.Option className={className} key='add'>
+        <div>{addNewContent || <><Antd.Icon type='plus' /> Add new</>}</div>
+      </Antd.Select.Option>
+    );
   }
 
-  private onChange (value: any) {
-    const foundOption = this.options.find(option => option.id === value.key);
+  private renderOptionEmpty () {
+    const { selectProps } = this.props
+      , className = `${SEARCH_CREATE_PREFIX_CLS}-item-empty`;
+
+    return (
+      <Antd.Select.Option className={className} disabled key='empty'>
+        <div>{get(selectProps, 'notFoundContent') || 'No results'}</div>
+      </Antd.Select.Option>
+    );
+  }
+
+  private renderOptionNoSearch () {
+    const { noSearchContent } = this.props
+      , className = `${SEARCH_CREATE_PREFIX_CLS}-item-empty`;
+
+    return (
+      <Antd.Select.Option className={className} disabled key='no-search'>
+        <div>{noSearchContent || 'Type in search text'}</div>
+      </Antd.Select.Option>
+    );
+  }
+
+  private renderOption (option: IEndpointOption) {
+    const className = `${SEARCH_CREATE_PREFIX_CLS}-item`;
+
+    return (
+      <Antd.Select.Option
+        className={className}
+        key={option.id}
+        value={option.id}
+      >
+        {option.name}
+      </Antd.Select.Option>
+    );
+  }
+
+  private onChange (selectedOption: any) {
+    if (selectedOption.key === 'add') {
```

Comment:
This hardcoding of keys is pointing to a code smell for me but it's the end of the day so I can't quite tease it out

Reply:
Well, It's a restriction where Ant will trigger an onChange no matter what. I was originally using an onClick, but onChange would trigger too and it was causing issues.

It might not be ideal, but I'm just catching the `add` item select in there.

Also, taking that route enables the keyboard whilst the onClick would not make it work.